File: backend/app/schemas/device.py
from pydantic import BaseModel, ConfigDict
from datetime import datetime
from typing import Literal, Optional

class DeviceResponse(BaseModel):
    id: int
    name: str
    location: str
    status: str
    last_seen_at: datetime | None
    created_at: datetime
    created_by: int
    model_config=ConfigDict(from_attributes=True)    

class DeviceCreate(BaseModel):
    name: str
    location: str
    status: Literal["offline", "online"] = "offline"

class DevicePatch(BaseModel):
    name: Optional[str] = None
    location: Optional[str] = None
    status: Optional[Literal["offline", "online"]] = None

File: frontend/app/types/device.ts
export type DeviceStatus = "offline" | "online"

export interface DeviceResponse {
    id: number
    name: string
    location: string
    status: DeviceStatus
    last_seen_at?: string
    created_at: string
    created_by: number
}

export interface DeviceCreate {
    name: string
    location: string
    status?: DeviceStatus
}

export interface DevicePatch {
    name?: string
    location?: string
    status?: DeviceStatus
}